backend/tracim_backend/lib/core/live_messages.py

```diff
@@ -66,9 +67,21 @@ def __init__(
             if not _grip_pub_control:
                 # NOTE S.G. - 2020-08-06 - publishing using ZMQ to avoid
                 # low maximum (1Mbytes) message size when using HTTP transport (#3415)
-                _grip_pub_control = GripPubControl(
-                    {"control_zmq_uri": config.LIVE_MESSAGES__CONTROL_ZMQ_URI}
-                )
+                _grip_pub_control = GripPubControl()
+
+                for zmq_control_uri, zmq_push_uri, zmq_pub_uri in zip_longest(
+                    config.LIVE_MESSAGES__CONTROL_ZMQ_URI,
+                    config.LIVE_MESSAGES__PUSH_ZMQ_URI,
+                    config.LIVE_MESSAGES__PUB_ZMQ_URI,
+                    fillvalue=None,
```

Comment:
Is there a reason for having different number of values?

- if yes: then add a comment in order to explain in which case
- if no: add an assert checking same number of values in `_load_live_messages_config()` so that tracim exits in case of wrong config

This will make tracim more robust

Reply:
The reason there is different number of values is the following:

GripControl will automatically resolve the `LIVE_MESSAGES__PUSH_ZMQ_URI` and `LIVE_MESSAGES__PUB_ZMQ_URI` based on `LIVE_MESSAGES__CONTROL_ZMQ_URI`.

The only moment it is required to fill these values is if the Pushpin server is not accessible by the ports defined in pushpin config (for example when being Docker port mappings). So the last two variables are not required all the time.

Reply:
After some thinking and after a small talk with @Skylsmoi I would like we make configuration more clear. There are only two cases to manage:

- you configure 3 parameters for all servers
- you configure only first parameter for all servers

Other cases are minor optiminisation and would make debug more difficult

Reply:
@lebouquetin ok noted I will change that. This means I should check if the other lists length is either `0` or `len(LIVE_MESSAGES__CONTROL_ZMQ_URI)` am I right ?

Reply:
done in last commit